modules/UI/UI.js

```diff
@@ -1365,7 +1368,14 @@ const UIListeners = new Map([
         UI.toggleChat
     ], [
         UIEvents.TOGGLE_SETTINGS,
-        () => UI.toggleSidePanel("settings_container")
+        () => {
+            if (interfaceConfig.SETTINGS_SECTIONS.length === 1
+                    && UIUtil.isSettingEnabled('devices')) {
```

Comment:
We should add some jsdoc explaining that we handle this case separately, because it's a dialog etc. etc.

Not related to the code above, but maybe while we're here we should also handle the case when no settings are listed in interfaceConfig. One may want to disable everything and then the settings button should not appear.